scipion/scripts/kickoff.py

```diff
@@ -421,33 +436,47 @@ def getTemplate(root):
         If more than one template is found or passed, a dialog is raised
         to choose one.
     """
-    templates = []
     templateFolder = getExternalJsonTemplates()
     customTemplates = len(sys.argv) > 1
     if customTemplates:
+        tl = TemplateList()
+        templates = []
         candidates = sys.argv[1:]
         for candFile in candidates:
             if os.path.isfile(candFile):
                 templates.append(String(candFile))
             else:
                 print(" > %s file does not exist." % candFile)
+        templates = tl.genFromStrList(templates).templates
     else:
+        # Check if other plugins have json.templates
+        domain = pw.Config.getDomain()
+        tempList = TemplateList()
```

Comment:
Clean code suggestion: t1 (line 442) and tempList (line 454) generates the same object in both if and else ways. It should be generated before the if statement.